pytorch_lightning/accelerators/cpu.py

```diff
@@ -39,19 +39,17 @@ def get_device_stats(self, device: _DEVICE) -> Dict[str, Any]:
         return {}
 
     @staticmethod
-    def parse_devices(devices: Union[int, str, List[int]]) -> Union[int, str, List[int]]:
+    def parse_devices(devices: Union[int, str, List[int]]) -> int:
         """Accelerator device parsing logic."""
+        devices = device_parser.parse_cpu_cores(devices)
+        assert devices is not None
```

Comment:
If `Optional[int]` is fixed from the `parse_cpu_cores` then we wouldn't need this assertion